netbox/utilities/jsonschema.py

```diff
@@ -0,0 +1,133 @@
+
+__all__ = (
+    'JSONSchemaProperty',
+    'PropertyTypeEnum',
+    'StringFormatEnum',
+)
+
+
+class PropertyTypeEnum(Enum):
+    STRING = 'string'
+    INTEGER = 'integer'
+    NUMBER = 'number'
+    BOOLEAN = 'boolean'
+    ARRAY = 'array'
+    OBJECT = 'object'
+
+
+class StringFormatEnum(Enum):
+    EMAIL = 'email'
+    URI = 'uri'
+    IRI = 'iri'
+    UUID = 'uuid'
+    DATE = 'date'
+    TIME = 'time'
+    DATETIME = 'datetime'
+
+
+FORM_FIELDS = {
+    PropertyTypeEnum.STRING.value: forms.CharField,
+    PropertyTypeEnum.INTEGER.value: forms.IntegerField,
+    PropertyTypeEnum.NUMBER.value: forms.FloatField,
+    PropertyTypeEnum.BOOLEAN.value: forms.BooleanField,
+    PropertyTypeEnum.ARRAY.value: SimpleArrayField,
+    PropertyTypeEnum.OBJECT.value: forms.JSONField,
+}
+
+STRING_FORM_FIELDS = {
+    StringFormatEnum.EMAIL.value: forms.EmailField,
+    StringFormatEnum.URI.value: forms.URLField,
+    StringFormatEnum.IRI.value: forms.URLField,
+    StringFormatEnum.UUID.value: forms.UUIDField,
+    StringFormatEnum.DATE.value: forms.DateField,
+    StringFormatEnum.TIME.value: forms.TimeField,
+    StringFormatEnum.DATETIME.value: forms.DateTimeField,
+}
+
+
+@dataclass
+class JSONSchemaProperty:
+    type: PropertyTypeEnum = PropertyTypeEnum.STRING.value
+    title: str | None = None
+    description: str | None = None
+    default: Any = None
+    enum: list | None = None
+
+    # Strings
+    minLength: int | None = None
+    maxLength: int | None = None
+    pattern: str | None = None  # Regex
+    format: StringFormatEnum | None = None
+
+    # Numbers
+    minimum: int | float | None = None
+    maximum: int | float | None = None
+    multipleOf: int | float | None = None
+
+    def to_form_field(self, name, required=False):
+        """
+        Instantiate and return a Django form field suitable for editing the property's value.
+        """
+        field_kwargs = {
+            'label': self.title or title(name),
+            'help_text': self.description,
+            'required': required,
+            'initial': self.default,
+        }
+
+        # Choices
+        if self.enum:
+            field_kwargs['choices'] = [(v, v) for v in self.enum]
+
+        # String validation
+        elif self.type == PropertyTypeEnum.STRING:
+            if self.minLength is not None:
+                field_kwargs['min_length'] = self.minLength
+            if self.maxLength is not None:
+                field_kwargs['max_length'] = self.maxLength
+            if self.pattern is not None:
+                field_kwargs['validators'] = [
+                    RegexValidator(regex=self.pattern)
+                ]
+
+        # Integer/number validation
+        elif self.type in (PropertyTypeEnum.INTEGER, PropertyTypeEnum.NUMBER):
+            if self.minimum:
+                field_kwargs['min_value'] = self.minimum
+            if self.maximum:
+                field_kwargs['min_value'] = self.maximum
```

Comment:
I feel like this should be assigning to `field_kwargs['min_value']` but somehow it still seems to work correctly, which leads me to believe that this validation is not actually being applied correctly.